logstash-core/spec/logstash/agent/converge_spec.rb

```diff
@@ -74,19 +74,18 @@
         describe "#running_user_defined_pipelines" do
           it "returns the user defined pipelines" do
             start_agent(subject)
-            subject.with_running_user_defined_pipelines do |pipelines|
-              expect(pipelines).to include(:main)
-              expect(pipelines).not_to include(:system_pipeline)
-            end
-            subject.shutdown
+            wait_for do
```

Comment:
Does `wait_for` have a timeout limit?

If not can you wrap this in a time out? 

Similar to here: https://github.com/elastic/logstash/pull/7752/files